### PR TITLE
Translate 2.4.0-preview1 release post (vi)

### PR DESCRIPTION
**Pull request by taduyanh**

### CODE REVIEW

**Comment by stomar** [comment]
Thanks for your translation, but this can not be merged like this.

* file name and time stamp **must** be the same as in `en` version
* please keep empty lines after headings, and also same markdown as in `en` version (e.g. in download section)
* do not break links across multiple lines, i.e. `[...](...)`

Please amend your commit and force push again (or open new PR if too complicated).

**Comment by taduyanh** [comment]
@stomar Thank you for your feedback, I've updated the translation, please take a look again

File: vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+---
+layout: news_post
+title: "Phát hành Ruby 2.4.0-preview1"
+author: "naruse"
+translator: "Tạ Duy Anh"
+date: 2016-06-21 11:00:00 +0700
```
Comment (stomar) [inline vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: wrong time stamp

File: vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+lang: vi
+---
+
+Chúng tôi rất hân hạnh được thông báo về việc phát hành Ruby phiên bản
+2.4.0-preview1.
+
+Phiên bản Ruby 2.4.0-preview1 là phiên bản preview đầu tiên của Ruby 2.4.9.
```
Comment (stomar) [inline vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: 2.4.0

File: vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+Bản preview1 lần này được phát hành sớm hơn so với thông thường vì nó bao gồm rất
+nhiều chức năng và cải tiến. Mọi người đừng ngại
+[phản hồi](https://bugs.ruby-lang.org/projects/ruby/wiki/HowToReport) lại cho
+chúng tôi về phiên bản này, vì các chức năng vẫn có thể được thay đổi và cập nhật.
+
+## [Hợp nhất Fixnum và Bignum vào Integer](https://bugs.ruby-lang.org/issues/12005)
+
+Mặc dù chuẩn [ISO/IEC 30170:2012](http://www.iso.org/iso/iso_catalogue/catalogue_tc/
```
Comment (stomar) [inline vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: link should not be wrapped over multiple lines

File: vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+Ruby 2.4 cũng bao gồm các cải thiện về hiệu năng song song với các thay đổi
+về chức năng. Sau đây là nội dung của các cải thiện:
+
+### [Array#max, Array#min](https://bugs.ruby-lang.org/issues/12172)
+
+`[x, y].max` và `[x, y].min` đã được tinh chỉnh lại để ko tạo các mảng
+tạm (temporary array) trong một số  trường hợp.
+
+### [Regex#match?](https://bugs.ruby-lang.org/issues/8110)
```
Comment (stomar) [inline vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: Regexp

File: vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+## Debugging
+
+### [Thread#report_on_exception và Thread.report_on_exception](https://bugs.ruby-lang.org/issues/6647)
+
+Ruby sẽ bỏ qua các exception trong các thread trừ khi có một thread khác tham
+gia vào tiến trình của thread có exception. Tuy nhiên với việc
+set `report_on_exception = true`, bạn có thể nhận biết được việc thread bị
+kết thúc bởi một exception chưa  được xử lý.
+Hãy góp ý thêm cho chúng tôi về giá trị mặc định của `report_on_exception`
```
Comment (stomar) [inline vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: new paragraph

File: vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+và về việc report-on-GC(thống kê ở Garbage Collection).
```
Comment (stomar) [inline vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: missing space before "(" ?

File: vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+Ruby đã có cơ chế phát hiện deadlock của các threads, tuy nhiên các thông tin
+được báo cáo về deadlock thường không đầy đủ cho việc debug. Từ bản 2.4 trở đi,
+khi phát hiện ra Deadlock, Ruby sẽ hiển thị các thread cùng với backtrace
+và các threads liên quan.
+
+Mời mọi người dùng thử và cảm nhận việc lập trình với Ruby 2.4.0-preview1,
+đồng thời [gửi phản hồi cho chúng tôi](https://bugs.ruby-lang.org/projects/ruby/wiki/HowToReport)!
+
+## Thay đổi đáng chú ý so với phiên bản 2.3
```
Comment (stomar) [inline vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: missing empty line

File: vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+Chi tiết mời xem [Thông tin](https://github.com/ruby/ruby/blob/v2_4_0_preview1/NEWS) và
+[Changelog](https://github.com/ruby/ruby/blob/v2_4_0_preview1/ChangeLog).
+Với lần cập nhật này, [1140 files đã được cập nhật, bao gồm 33126 insertion và 50933 deletions so với phiên bản 2.3.0](https://github.com/ruby/ruby/compare/v2_3_0...v2_4_0_preview1)!
```
Comment (stomar) [inline vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: should be new paragraph

File: vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+## Tải về
+
```
Comment (stomar) [inline vi/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: download link section should exactly match `en` version (no diff)

**Comment by joneslee85** [comment]
thanks @taduyanh

**Pull request merged**